plugins/plugin-chart-echarts/src/Gauge/controlPanel.tsx

```diff
@@ -0,0 +1,267 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { t, validateNonEmpty } from '@superset-ui/core';
+import { sharedControls, ControlPanelConfig, D3_FORMAT_OPTIONS } from '@superset-ui/chart-controls';
+import { DEFAULT_FORM_DATA } from './types';
+
+const {
+  title,
+  minVal,
+  maxVal,
+  fontSize,
+  numberFormat,
+  roundcap,
+  showSplitLine,
+  showAxisTick,
+  valueName,
+  valueFormatter,
+  valueAnimation,
+  startAngle,
+  endAngle,
+  intervals,
+  intervalColorIndices,
+} = DEFAULT_FORM_DATA;
+
+const config: ControlPanelConfig = {
+  controlPanelSections: [
+    {
+      label: t('Query'),
+      expanded: true,
+      controlSetRows: [
+        ['groupby'],
+        ['metric'],
+        ['adhoc_filters'],
+        ['row_limit', null],
+        [
+          {
+            name: 'min_val',
+            config: {
+              ...sharedControls.metric,
+              type: 'MetricsControl',
```

Comment:
@krsnik93 Thanks for your contribution,
please do not direct uses 'MetricsControl`, when `ENABLE_EXPLORE_DRAG_AND_DROP` is enabled(from config.py), the control type will be dynamically selected.

Reply:
I wanted to give the option of having a query result as either the start or the end of the axis. For example, user sets `min_val` as `0`, `max_val` as `sum(some_column)` and then sets `metric` as `avg(some_column)`. If you think this is overengineering, I can replace as suggested above.

Reply:
I can understand the intention but this might not be necessary even from a user-experience point of view. Having a dynamic range could be useful in some cases, but for most cases, as a dashboard consumer, it'd be confusing to see the goal changes from time to time. The purpose of a gauge chart is normally for monitoring the progress towards a certain goal; if the goal is dynamic, then it's a little bit difficult to reason whether the progress is satisfying or not.

Considering it's very easy to change the bounds and no other charts have similar capability, I'd vote for leave it to chart builders to manually manage the bounds.